nodespec/azure.go

```diff
@@ -0,0 +1,263 @@
+  maxDataDiskCount: 16
+  memoryInMb: 68719.476736
+  name: Standard_E8as_v4
+  numberOfCores: 8
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 137438.953472
+Standard_E8s_v3:
+  description: Esv3-series, memory optimized, 160-190 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 16
+  memoryInMb: 68719.476736
+  name: Standard_E8s_v3
+  numberOfCores: 8
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 137438.953472
+Standard_E16s_v3:
+  descriptions: Esv3-series, memory optimized, 160-190 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 137438.953472
+  name: Standard_E16s_v3
+  numberOfCores: 16
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 274877.906944
+Standard_E32s_v3:
+  description: Esv3-series, memory optimized, 160-190 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 274877.906944
+  name: Standard_E32s_v3
+  numberOfCores: 32
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 549755.813888
+Standard_F4s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 8
+  memoryInMb: 8589.934592
+  name: Standard_F4s_v2
+  numberOfCores: 4
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 34359.738368
+Standard_F8s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 16
+  memoryInMb: 17179.869184
+  name: Standard_F8s_v2
+  numberOfCores: 8
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 68719.476736
+Standard_F16s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 34359.738368
+  name: Standard_F16s_v2
+  numberOfCores: 16
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 137438.953472
+Standard_F32s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 68719.476736
+  name: Standard_F32s_v2
+  numberOfCores: 32
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 274877.906944
+`
+)
+
+// ProviderAzure contains all provider specific info
+type ProviderAzure struct {
+	vmSizes map[string]VMSize
+}
+
+type VMSize struct {
+	Description          string  `yaml:"description"`
+	MaxDataDiskCount     int     `yaml:"maxDataDiskCount"`
+	MemoryInMB           float64 `yaml:"memoryInMb"`
+	Name                 string  `yaml:"name"`
+	NumberOfCores        int64   `yaml:"numberOfCores"`
+	OSDiskSizeInMB       int64   `yaml:"osDiskSizeInMb"`
+	ResourceDiskSizeInMB float64 `yaml:"resourceDiskSizeInMb"`
+}
+
+// NewAzureProvider initiates a new Azure provider with the information about VM sizes.
+func NewAzureProvider() (*ProviderAzure, error) {
+	p := &ProviderAzure{}
+
+	err := yaml.Unmarshal([]byte(azureVMSizesYAML), &p.vmSizes)
```

Comment:
I wonder if we actually need to store them in YAML. I mean, it's easy to just copy/paste the VM Size from the installations repo, but it's very fragile...
Wouldn't we be better off storing them in a `map[string]VMSize`? It would definitely speed things up, since we wouldn't have to unmarshal the YAML

Reply:
I just went the AWS-way as it was the quickest :grin:

Manually translating YAML to `map[string]VMSize` could be error prone maybe. Ideally we would fetch VM sizes data from a single place, wherever we need it.